bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,256 @@
+            params=plasmod_options["params"],
+            build_config=plasmod_options["build_config"],
+        )
+        plasmod_solver.execute()
+
+        if plot:
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.pprime(plasmod_solver.x), "pprime", "-"
+            )
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.ffprime(plasmod_solver.x), "ffrime", "-"
+            )
+
+        # generate mesh for the Grad-Shafranov solver
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_mesh,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {
+            "lcar": lcar_mesh,
+            "physical_group": "plasma_face",
+        }
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )[0]
+
+        # initialize the Grad-Shafranov solver
+        p_order = gs_options["p_order"]
+        gs_solver = FemGradShafranovFixedBoundary(mesh, p_order=p_order)
+
+        bluemira_print("Solving fixed boundary Grad-Shafranov...")
+
+        gs_solver.solve(
+            plasmod_solver.pprime,
+            plasmod_solver.ffprime,
+            plasmod_solver.I_p,
+            iter_err_max=gs_options["iter_err_max"],
+            max_iter=gs_options["max_iter"],
+            relaxation=gs_options["relaxation"],
+            plot=gs_options["plot"],
+        )
+
+        _, kappa_95, delta_95 = calculate_plasma_shape_params(
+            gs_solver.psi_norm_2d,
+            mesh,
+            np.sqrt(0.95),
+        )
+
+        # calculate the iteration error
+        err_delta = abs(delta_95 - delta95_t) / delta95_t
+        err_kappa = abs(kappa_95 - kappa95_t) / kappa95_t
+        iter_err = max(err_delta, err_kappa)
+
+        # calculate the new kappa_u and delta_u
+        kappa_u_0 = builder_plasma.params.get_param("kappa_u")
+        delta_u_0 = builder_plasma.params.get_param("delta_u")
+
+        kappa_u = (1 - relaxation) * kappa_u_0 * (
+            kappa95_t / kappa_95
+        ) + relaxation * kappa_u_0
+        delta_u = (1 - relaxation) * delta_u_0 * (
+            delta95_t / delta_95
+        ) + relaxation * delta_u_0
+
+        if verbose:
+            print("previous shape parameters")
+            print(f"{kappa_u_0}, {delta_u_0}")
+
+            print("recalculated shape parameters")
+            print(f"{kappa_u}, {delta_u}")
+
+            print(" ")
+            print(f"bluemira delta95 = {delta_95}")
+            print(f"target delta95 = {delta95_t}")
+
+            print(f"|Target - bluemira|/Target = {err_delta}")
+
+            print(" ")
+            print(f"bluemira kappa95 = {kappa_95}")
+            print(f"target kappa95 = {kappa95_t}")
+
+            print(f"|Target - bluemira|/Target = {err_kappa}")
```

Comment:
Perhaps this is a little too verbose even for `verbose=True`? With the `bluemira_debug` this is largely already captured. There is a lot going on in terms of printing to the console at the moment, perhaps as this emerges from development we could look to condense the output a little more than I have already done?